### ISSUE TITLE
how to leave a room / disconnect / stop listening?

### ISSUE THREAD

**Issue by dcsan** [gitterHQ/node-gitter#15]
the streaming API will allow me to connect, but it seems to remember connections between different instances of my app. I'm getting multiple instances of a message sent to me.

Is there a way to disconnect?
Do i have to revoke my apps API key?

https://developer.gitter.im/docs/streaming-api
https://github.com/gitterHQ/gitter-node-stream/blob/master/gitter-stream.js

edit: this seems to have cleared up, maybe there was a delay for unsubscribes? either way would be nice to have an api for. I know it's possible to listen twice to the same room, as I've seen that behavior.

ps I'm using 
```
        var chats = room.streaming().chatMessages();
        chats.on('chatMessages', function(message) {
...
```

**Comment by dcsan** [comment]
this was a test mode server issue

**Comment by dcsan** [comment]
reopening. seems like an intermittent problem, and not sure the root of it. 

rebooting my own app has no effect, it seems the subscriptions to events are applied on the server, so working on an app that frequently restarts (eg running tests) seems to over-subscribe me to events, such that any message gets multiple `on` events fired to me.

![image](https://cloud.githubusercontent.com/assets/514002/8899593/34716d28-33eb-11e5-8757-1dde8a005089.png)

**Comment by mydigitalself** [comment]
@dcsan are you able to show us any code?

**Comment by dcsan** [comment]
@mydigitalself thanks for the quick response. I added you to the private repo
https://github.com/dcsan/gitterbot/tree/master/nap

joining rooms is here:
https://github.com/dcsan/gitterbot/blob/4-multiple-joins/nap/lib/bot/GBot.js#L267-L282

using GitterHelper


on running my tests, which does connect to the API, I'm sometimes seeing this:
```
Not possible to join the room:  [Error: 429 {"error":"API rate limit exceeded"}] bothelp
Not possible to join the room:  [Error: 429 {"error":"API rate limit exceeded"}] bothelp/GeneralChat
Not possible to join the room:  [Error: 429 {"error":"API rate limit exceeded"}] bothelp/JS-Basics
Not possible to join the room:  [Error: 429 {"error":"API rate limit exceeded"}] bothelp/Frontend
```

and then seeing lots of dupe echoes.

then running tests a couple mins later i see:

```
Not possible to join the room:  [Error: 403 {"error":"Permission denied"}] bothelp
```
and things are back to normal...


is there a way with the API to see if i'm already listening to events for that room?

**Comment by dcsan** [comment]
closing again! I think this is due to my testrunner running tests in parallel or something... not 100% sure but I cant repro in the app atm and don't want to waste your time til I can identify the repro steps.
